### ISSUE TITLE
Resurrect and reformat the \dataset markup for inline DOI citations

### ISSUE THREAD

**Issue by augustfly** [AASJournals/AASTeX60#5]
For authors who wish to provide inline links to data that should not otherwise be cited, e.g., raw data; e.g., MAST data collections, rewrite [the current \dataset AASTeX6 function](https://github.com/AASJournals/AASTeX60/blob/master/cls/aastex6.cls#L1082) to take advantage of our use of hyperref and provide a "DATASET" inline link authors can markup in their manuscripts. 

The latex markup:

> We used HST data \dataset{doi} to find...

would produce the following inline 

> We used HST data [DATALINK] to find...

The underlying markup should be something like

> \href{#1}{[DATALINK]}

**Comment by gregschwarz** [comment]
Let me make sure I understand what you are proposing. You want to add hypertext mark up to \dataset?  This should be possible since \dataset already takes two arguments.  The {} argument was what is shown in the text. The [] argument was meant to be reserved for a dataset ID and not shown. The publisher would establish the link during production after verification through ADS.  It would certainly make sense to just add in \url to the [] argument to have the {} text be a live link.  Is that in essences what you are proposing?

**Comment by augustfly** [comment]
Hmm, making me think again. 

First, yes, I am proposing `\dataset` to be rewritten using `\href` to make it a hypertext link. That way the reader can click through directly to the data link inline to the article text. 

Second, I assumed that we wouldn't give the user anyway to change the link's appearance inline. The reason for this was to standardize the typesetting of  "[DATALINK]" as a hypertext link to some kind of data. The reader doesn't need to see the URL inline, in fact that is terrible typesetting, but they should have a clear hint that whenever they see that [DATALINK] it will take them somewhere actionable. This is a change from the old `\dataset` markup which was just a semantic tag around a "URI" not a URL.  

This means the author's markup would simplify to \dataset{DOI-URL} to would create a [DATALINK] inline to the article. By DOI-URL, I mean an http DOI string not just a DOI URI.

From a revising aastex6.cls point of view the renewcommand would look something like

> \href{#1}{[DATALINK]}

since the 2nd `\href` argument is the anchor text for the 1st element URL. This argument ordering is flipped from the current way aastex6 `\dataset` is setup, where the first argument is the anchor text for the 2nd argument dataset tag.  I think this backwards _incompatibility_ is okay since basically no one ever uses `\dataset` anymore. If they did use the old format then it only breaks if they give an anchor text, and that can be fixed by hand. 

From the data editors POV we would just grok on that tag to double check them, and use this markup to standardize how DOIs to non-citable data get marked up.

From IOP's point of view they could add a typesetting layer where they double check that the argument is an http URL instead of a URI. Moreover they would then be able to highlight these data links as we've been discussing all along. 

To summarize it gives us, for free, semantic tagging around an increasingly common form of data linking, resurrecting something we've tried before. 

I'm adding @aaccomazzi just so he sees some of this as it is basically a co-opting of datasetIDs for DOIs and our markup purposes.

**Comment by gregschwarz** [comment]
We might need to talk about this over the phone to nail down the details but it sounds like you basically want to alias \href{#1}{DATALINK} to \dataset{DATALINK} to standardize this functionality.

**Comment by augustfly** [comment]
strictly speaking the author latex

> we provide the data for this article \dataset{doi}

should render as would

> we provide the data for this article \href{doi}{[DATALINK]}

which would like this in the compiled version.

> we provide the data for this article [[DATALINK]](example.com)

**Comment by aaccomazzi** [comment]
I hereby give you the permission to do whatever you will with \dataset (it's your macros after all!).  What matters to us anyway is what we get in the fulltext XML.  In fact it would be useful for us to find out at some point how this gets encoded in the VOR.

Does this mean that the decision about citation vs. mention has been made (in favor of mention)?

**Comment by augustfly** [comment]
@aaccomazzi yes, for the MAST data collections they will only be mentions. there was a lot of negative energy around adding collection DOIs to the references with the argument being that they have no independent intellectual value. this is from MAST btw.

**Comment by augustfly** [comment]
Update in #7
